mongodb-odbc-driver/installer/dmg/scripts/postinstall

```diff
@@ -1,82 +1,176 @@
+MDB_ANSI="MongoDB ANSI ODBC $LATEST_VERSION"
+MDB_UNICODE_DRIVERS="$INSTALL_PATH/libmdbodbcw.so"
+MDB_ANSI_DRIVERS="$INSTALL_PATH/libmdbodbca.so"
+CREATED_NEW_INI_FILE=0
 
-if __name__ == '__main__':
-    main()
+# Create the ODBC directory if it doesn't exist already
+if [[ ! -e $ODBC_PATH ]]; then
+    info "Creating $ODBC_PATH directory"
+    mkdir -p $ODBC_PATH
+elif [[ ! -d $ODBC_PATH ]]; then
+     err "$ODBC_PATH already exists but is not a directory"
+     exit 1   
+fi
+
+#Update the odbcinst.ini with the latest driver path and mark it as installed
+if [[ ! -e $INI ]]; then
+    info "$INI does not exists, creating"
+    touch $INI
+    echo "[ODBC Drivers]" >> $INI
+    echo "$MDB_UNICODE = Installed" >> $INI
+    echo "$MDB_ANSI = Installed" >> $INI
+    echo "" >> $INI
+    echo "[$MDB_UNICODE]" >> $INI
+    echo "Driver = $MDB_UNICODE_DRIVERS" >> $INI
+    echo "" >> $INI
+    echo "[$MDB_ANSI]" >> $INI
+    echo "Driver = $MDB_ANSI_DRIVERS" >> $INI
+    $CREATED_NEW_INI_FILE=1
+fi
+
+#If there was an existing odbcinst file, update the odbcinst.ini
+# with the latest driver path
+if [[ $CREATED_NEW_INI_FILE == 0 ]]; then
+    BACK_INI="$INI.back"
+    info "Back up file $BACK_INI"
+    IN_ODBC_DRIVERS_SECTION=0
+    IN_MDB_UNICODE_DRIVER_SECTION=0
+    IN_MDB_ANSI_DRIVER_SECTION=0
+    MDB_UNICODE_DRIVER_INSTALLED=0
+    MDB_ANSI_DRIVER_INSTALLED=0
+    MDB_UNICODE_DRIVER_CONFIGURED=0
+    MDB_ANSI_DRIVER_CONFIGURED=0
+
+    # Make a copy of the existing ini file
+    cp $INI $BACK_INI
+    if [[ -e $BACK_INI ]]; then
+        # Clear the INI file
+        > $INI
+
+        # Read the backup odbcinst.ini.back file and recreate the new updated odbcinst.ini one
+        while IFS= read -r line
+        do
+            if [[ "$line" =~ ^\[.*\]$ ]]; then
+                # Found a section
+                if grep -iqF "[ODBC Drivers]" <<< "$line"; then
+                    # Found the [ODBC Drivers] section 
+                    # where all drivers are listed
+                    IN_ODBC_DRIVERS_SECTION=1
+                elif grep -iqF "[$MDB_UNICODE]" <<< "$line"; then
+                    # Found the [MongoDB Unicode ODBC {version}] section
+                    IN_MDB_UNICODE_DRIVER_SECTION=1
+                    IN_MDB_ANSI_DRIVER_SECTION=0
+                    IN_ODBC_DRIVERS_SECTION=0
+                    echo "IN_MDB_UNICODE_DRIVER_SECTION=1"
+                elif grep -iqF "[$MDB_ANSI]" <<< "$line"; then
+                    # Found the [MongoDB ANSI ODBC {version}] section
+                    IN_MDB_ANSI_DRIVER_SECTION=1
+                    IN_MDB_UNICODE_DRIVER_SECTION=0
+                    IN_ODBC_DRIVERS_SECTION=0
+                    echo "IN_MDB_ANSI_DRIVER_SECTION=1"
+                else
+                    # Found a new section right after [ODBC Drivers]
+                    # Need to add the drivers lines in [ODBC Drivers] section
+                    if [ $IN_ODBC_DRIVERS_SECTION == 1 ]; then
```

Comment:
Will there always be a section after [ODBC Drivers]?  If not, we might miss printing out the `Installed` lines?  We may need logic to determine we are at the end of file and have not printed `Installed`.